isis/src/base/objs/History/History.cpp

```diff
@@ -56,24 +59,19 @@ namespace Isis {
     p_history.addObject(obj);
   }
 
-  /**
-   *
-   */
-  void History::WriteInit() {
+  Blob *History::toBlob(const QString &name) {
     ostringstream ostr;
-    if (p_nbytes > 0) ostr << std::endl;
     ostr << p_history;
     string histStr = ostr.str();
-    int bytes = histStr.size();
+    int nbytes = histStr.size();
 
-    char *temp = p_buffer;
-    p_buffer = new char[p_nbytes+bytes];
-    if (temp != NULL) memcpy(p_buffer, temp, p_nbytes);
+    char *buffer = new char[nbytes];
```

Comment:
This buffer is a memory leak right now. If we instead pass ownership in Blob::setData this problem goes away.